### PR TITLE
adds simple mode input communication struct.

### PR DESCRIPTION
**Pull request by jdcaporale**
Adds a simple one integer mode input struct for switching modes in a state machine or set of behaviors.

### CODE REVIEW

**Comment by ethanmusser** [comment]
Looks good.

**Review by ethanmusser — APPROVED** [review]
Looks good.

File: python_scripts/mode_input_cli.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    print(print_msg)
+
+
+def header_str(off_int):
+    return """
++--------------------------------------------------------------------+
+| Enter a valid mode integer or 'q' to quit and return to OFF state. |
+| Inputting mode %d or Ctrl-C will send a kill command.              |
++--------------------------------------------------------------------+
+    """.format(off_int)
```
Comment (jdcaporale) [inline python_scripts/mode_input_cli.py]: Should be KILL not OFF int

File: python_scripts/mode_input_cli.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        while True:
+            mode = input(f'Enter mode:  ')
+            try:
+                publish_mode(int(mode))
+            except ValueError:
+                if mode.upper() == 'Q':
+                    publish_mode(OFF, 'Exiting.')
+                    break
+                else:
+                    publish_mode(OFF, 'Invalid input.  Exiting.')
```
Comment (jdcaporale) [inline python_scripts/mode_input_cli.py]: Do we want any error to turn off the robot?

File: python_scripts/mode_input_cli.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                    break
+            except AssertionError:
+                publish_mode(OFF, 'Invalid input.  Exiting.')
+                break
+    except KeyboardInterrupt:
+        publish_mode(KILL, 'Keyboard interrupt.  Killing Robot and exiting.')
```
Comment (jdcaporale) [inline python_scripts/mode_input_cli.py]: May want to send the kill command a few times to ensure it gets read

File: python_scripts/mode_input_cli.py (hunk 1 above)
Comment (ethanmusser) [reply python_scripts/mode_input_cli.py]: Fixed in 2e7357df5b608e0ef2f9850ad300cb9611ea49e6

**Review by jdcaporale — COMMENTED** [review]
LGTM now.

File: python_scripts/mode_input_cli.py (hunk 3 above)
Comment (ethanmusser) [reply python_scripts/mode_input_cli.py]: Fixed in 2e7357df5b608e0ef2f9850ad300cb9611ea49e6

**Pull request merged**